### PR TITLE
chore(android): put back manifest rejection on permissions calls

### PR DESCRIPTION
**Pull request by carlpoole**
This would need to be also handled when overriding `onRequestPermissionsResult` in a plugin.

### CODE REVIEW

File: android/capacitor/src/main/java/com/getcapacitor/Plugin.java [hunk 1]
```diff
@@ -602,6 +609,20 @@ public void handlePermissions(String[] permissions, int[] grantResults) {
...
+
+        if (!hasDefinedPermissions(permissions)) {
+            StringBuilder builder = new StringBuilder();
+            builder.append("Missing the following permissions in AndroidManifest.xml:\n");
+            String[] missing = PermissionHelper.getUndefinedPermissions(getContext(), permissions);
+            for (String perm : missing) {
+                builder.append(perm + "\n");
+            }
+            savedLastCall.reject(builder.toString());
+            savedLastCall = null;
```
Comment (jcesarmobile) [inline android/capacitor/src/main/java/com/getcapacitor/Plugin.java:621]:
```suggestion
```

don't set the call to null, let the code after the removed do it.

**Review by jcesarmobile — CHANGES_REQUESTED** [review]
Requested a few changes.

`checkPermissions` doesn't reject if the permissions are not there, but since it's a new method I'm ok with it. It will return prompt, so the user will try to request the permissions and then get the rejection on the code that request the permissions.

File: android/capacitor/src/main/java/com/getcapacitor/Plugin.java [hunk 2]
```diff
@@ -560,20 +560,27 @@ protected void onRequestPermissionsResult(int requestCode, String[] permissions,
             return;
         }
 
-        this.handlePermissions(permissions, grantResults);
+        if (validatePermissions(permissions, grantResults)) {
+            savedCall.resolve(getPermissionStates());
+        } else {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (carlpoole) [reply android/capacitor/src/main/java/com/getcapacitor/Plugin.java]: thanks, this is a lot cleaner. I applied your suggestion

**Review by jcesarmobile — APPROVED** [review]

**Comment by dwieeb** [comment]
> This would need to be also handled when overriding `onRequestPermissionsResult` in a plugin.

Why might a plugin need to do this?

**Comment by carlpoole** [comment]
> > This would need to be also handled when overriding `onRequestPermissionsResult` in a plugin.
> 
> Why might a plugin need to do this?

if its not handled when overriding, then the manifest check/rejection won't occur. Plugin authors can call `validatePermissions` to handle saving the plugin states and check if the manifest lacks the permissions in one go.

**Review by dwieeb — APPROVED** [review]

**Pull request merged**